osutil/disks/disks.go

```diff
@@ -336,59 +337,88 @@ func (d *disk) FindMatchingPartitionUUID(label string) (string, error) {
 	// if we haven't found the partitions for this disk yet, do that now
 	if d.fsLabelToPartUUID == nil {
 		d.fsLabelToPartUUID = make(map[string]string)
-		// step 1. find all devices with a matching major number
-		// step 2. start at the major + minor device for the disk, and iterate over
-		//         all devices that have a partition attribute, starting with the
-		//         device with major same as disk and minor equal to disk minor + 1
-		// step 3. if we hit a device that does not have a partition attribute, then
-		//         we hit another disk, and shall stop searching
-
-		// note that this code assumes that all contiguous major / minor devices
-		// belong to the same physical device, even with MBR and
-		// logical/extended partition nodes jumping to i.e. /dev/sd*5
-
-		// start with the minor + 1, since the major + minor of the disk we have
-		// itself is not a partition
-		currentMinor := d.minor
-		for {
-			currentMinor++
-			partMajMin := fmt.Sprintf("%d:%d", d.major, currentMinor)
-			props, err := udevProperties(filepath.Join("/dev/block", partMajMin))
-			if err != nil && strings.Contains(err.Error(), "Unknown device") {
-				// the device doesn't exist, we hit the end of the disk
-				break
-			} else if err != nil {
-				// some other error trying to get udev properties, we should fail
-				return "", fmt.Errorf("cannot get udev properties for partition %s: %v", partMajMin, err)
+
+		// step 1. find the devpath for the disk, then glob for matching
+		//         devices using the devname in that sysfs directory
```

Comment:
one can simply ReadDir of `filepath.EvalSymlinks(filepath.Join("/sys/dev/block", d.Dev()))` and test things to be real subdirs (nonsymlinks). These will be partitions of  a device.

This avoids iterating _all_ block devices.

Reply:
same as above, this improvement is good and correct and can be done in a followup PR

Reply:
Once we drop udev, maybe we should unify or move the other helper we have in the `gadget` package.